### PR TITLE
Backend release make

### PR DESCRIPTION
**Pull request by natalie-elizabeth**
#### What Does This PR Do?

- Updates Circleci config.yml file for deployment.
- Adds deployment scripts using `deployBackend.sh`.
- Updates makefile with shortcuts for backend deployment, tagging and publishing.

#### How can this be manually tested?
For building the backend application run `make build_backend`. To tag and publish run `make tag` and `make publish` respectively.

### CODE REVIEW

File: makefile [hunk 1]
```diff
@@ -34,6 +45,61 @@ stop:
 	@ docker-compose -f $(DOCKER_COMPOSE_FILE) down -v
 	@ echo "All containers stopped successfully"
 
+build_backend:
```
Comment (andela-sjames) [inline makefile]: Do we have a readme doc for these changes?

File: scripts/deployBackend.sh [hunk 2]
```diff
@@ -0,0 +1,82 @@
+#!/bin/bash
+set -o errexit
+set -o pipefail
+
+set_variables() {
+    COMMIT_HASH=$(git rev-parse --short HEAD)
```
Comment (andela-sjames) [inline scripts/deployBackend.sh]: Readme should be updated for this too

Comment (natalie-elizabeth) [reply scripts/deployBackend.sh]: Will update the README.

File: makefile (hunk 1 above)
Comment (natalie-elizabeth) [reply makefile]: ReadMe doc updated.

File: docker/release/client/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,19 @@
+FROM node:carbon
+
+LABEL MAINTAINER="Natalie Elizabeth <natalie.elizabeth@andela.com>"
+LABEL application="asocialsbackend"
```
Comment (andela-sjames) [inline docker/release/client/Dockerfile]: backend label for frontend dockerfile?

**Pull request merged**